lib/src/divider.dart

```diff
@@ -0,0 +1,360 @@
+/// Internal interface to the Divider.
+class MultiCycleDividerInterface extends PairInterface {
+  /// Clock for sequential logic.
+  Logic get clk => port('clk');
+
+  /// Reset for sequential logic (active high).
+  Logic get reset => port('reset');
+
+  /// Dividend (numerator) for the division operation.
+  Logic get dividend => port('dividend');
+
+  /// Divisor (denominator) for the division operation.
+  Logic get divisor => port('divisor');
+
+  /// The integrating environment is ready to accept the output of the module.
+  Logic get readyOut => port('readyOut');
+
+  /// Request for a new division operation to be performed.
+  Logic get validIn => port('validIn');
+
+  /// Quotient (result) for the division operation.
+  Logic get quotient => port('quotient');
+
+  /// A Division by zero occurred.
+  Logic get divZero => port('divZero');
+
+  /// The result of the current division operation is ready.
+  Logic get validOut => port('validOut');
+
+  /// The module is ready to accept new inputs.
+  Logic get readyIn => port('readyIn');
+
+  /// The width of the data operands and result.
+  final int dataWidth;
+
+  /// A constructor for the divider interface.
+  MultiCycleDividerInterface({this.dataWidth = 32})
+      : super(portsFromProvider: [
+          Port('clk'),
+          Port('reset'),
+          Port('dividend', dataWidth),
+          Port('divisor', dataWidth),
+          Port('validIn'),
+          Port('readyOut'),
+        ], portsFromConsumer: [
+          Port('quotient', dataWidth),
+          Port('divZero'),
+          Port('validOut'),
+          Port('readyIn'),
+        ]);
+
+  /// A match constructor for the divider interface.
+  MultiCycleDividerInterface.match(MultiCycleDividerInterface other)
+      : this(dataWidth: other.dataWidth);
+}
+
+/// The Divider module definition
+class MultiCycleDivider extends Module {
+  /// The Divider's interface declaration.
+  late final MultiCycleDividerInterface intf;
+
+  /// The width of the data operands and result.
+  late final int dataWidth;
+
+  /// The log of the data width representing
+  /// the number of bits required to store that number.
+  late final int logDataWidth;
+
+  /// The Divider module's constructor
+  MultiCycleDivider(MultiCycleDividerInterface interface)
+      : dataWidth = interface.dataWidth,
+        logDataWidth = log2Ceil(interface.dataWidth),
+        super(name: 'divider') {
+    intf = MultiCycleDividerInterface.match(interface)
+      ..pairConnectIO(
+        this,
+        interface,
+        PairRole.consumer,
+        uniquify: (original) => '${super.name}_$original',
+      );
+
+    _build();
+  }
+
+  /// Named constructor that explicitly takes Logics instead of an Interface.
+  MultiCycleDivider.ofLogics({
+    required Logic clk,
+    required Logic reset,
+    required Logic validIn,
+    required Logic dividend,
+    required Logic divisor,
+    required Logic readyOut,
+    required Logic quotient,
+    required Logic divZero,
```

Comment:
Only the inputs need to be arguments of the constructor, we can make the outputs just normal `get`ters of those output ports?

Reply:
Ah OK. I think this should be fixed now? Does the alternate constructor otherwise look good?

Reply:
Specifically I was unsure if I need to do anything when I instantiate the interface to indicate its PairRole, uniquifying the port names, etc.